osf/management/commands/populate_impact_metrics.py

```diff
@@ -0,0 +1,92 @@
+)
+
+from osf.models import Institution, OSFUser, Preprint
+
+
+# Edit the following guids based off your local environment
+USERS = [OSFUser.load('abd53u'), OSFUser.load('hy84n'), OSFUser.load('7zyg2')]
+PREPRINTS = [Preprint.load('far32'), Preprint.load('67bzg')]
+INSTITUTION = Institution.load('cos')
+
+def populate_preprint_metrics(dates):
+    for date in dates:
+        for user in USERS:
+            for preprint in PREPRINTS:
+                preprint_view_count = int(10 * random())
+                PreprintView.record_for_preprint(
+                    preprint=preprint,
+                    user=user,
+                    path=preprint.primary_file.path,
+                    timestamp=date,
+                    count=preprint_view_count
+                )
+
+                preprint_download_count = int(10 * random())
+                PreprintDownload.record_for_preprint(
+                    preprint=preprint,
+                    user=user,
+                    path=preprint.primary_file.path,
+                    timestamp=date,
+                    count=preprint_download_count
+                )
+
+
+def populate_institution_metrics(dates):
+    institution_public_project_count = 25 + (int(25 * random()))
+    institution_private_project_count = 25 + (int(25 * random()))
+    for date in dates:
+        institution_public_project_count = institution_public_project_count - int(5 * random())
+        institution_private_project_count = institution_private_project_count - int(5 * random())
+
+        InstitutionProjectCounts.record_institution_project_counts(
+            institution=INSTITUTION,
+            public_project_count=institution_public_project_count,
+            private_project_count=institution_private_project_count,
+            timestamp=date
+        )
+
+        for user in USERS:
+            user_public_project_count = (int(10 * random()))
+            user_private_project_count = (int(10 * random()))
+
+            UserInstitutionProjectCounts.record_user_institution_project_counts(
+                institution=INSTITUTION,
+                user=user,
+                public_project_count=user_public_project_count,
+                private_project_count=user_private_project_count,
+                timestamp=date
+            )
+
+
+class Command(BaseCommand):
+
+    def add_arguments(self, parser):
+        super(Command, self).add_arguments(parser)
+        parser.add_argument(
+            '--institutions',
+            action='store_true',
+            help='Only populate institutions related date'
+        )
+        parser.add_argument(
+            '--preprints',
+            action='store_true',
+            help='Only populate preprints related date'
+        )
+
+    def handle(self, *args, **options):
+        today = dt.datetime.today()
+        last_seven_days = [(today - dt.timedelta(days=num_days)) for num_days in range(0, 7)]
+
+        if not options['preprints']:
```

Comment:
This will get weird when we add more things, and I expect we're going to be using impact a lot in the future, so we'll have plenty of more things to add. Would it be better to just have a parameter for which one to run, or to maybe have a script per type of thing you want to add? There's not a lot of overlap between the code, so there doesn't seem to be a lot of benefit to combining these into one script.

Reply:
I think splitting it up makes the most sense and will do so.